### PR TITLE
Allow some configuration of the server

### PR DESCRIPTION
**Pull request by smondet**
This allows to mitigate issues like #42 in environments where Kubernetes
can be a bit unreliable.

### CODE REVIEW

File: src/lib/command_line.ml [hunk 1]
```diff
@@ -78,15 +83,47 @@ let main () =
...
     $ required_string "cluster-zone" (fun s -> `Zone s)
-      ~doc:"Zone of the Kubernetes cluster"
+      ~doc:"Zone of the Kubernetes cluster."
     $ begin
       pure (fun s -> `Max_nodes s)
       $ Arg.(
           required & opt (some int) None
           & info ["max-nodes"]
-            ~doc:"Maximum number of nodes in th cluster" ~docv:"NUMBER")
+            ~doc:"Maximum number of nodes in th cluster." ~docv:"NUMBER")
```
Comment (ihodes) [inline src/lib/command_line.ml]: the*

File: src/lib/command_line.ml [hunk 2]
```diff
@@ -78,15 +83,47 @@ let main () =
...
+      (`Min_sleep min_sleep)
+      (`Max_sleep max_sleep) ->
+      Server.Configuration.make ()
+        ~min_sleep ~max_sleep ~max_update_errors
+    end
+    $ begin
+      pure (fun s -> `Max_update_errors s)
+        $ Arg.(value & opt int Server.Configuration.Default.max_update_errors &
+               info ["max-update-errors"]
+                 ~doc:"The number of `kubectl` errors allowed before \
```
Comment (ihodes) [inline src/lib/command_line.ml]: Might be nice to have the default in the doc.

Comment (smondet) [inline src/lib/command_line.ml]: Cmdliner does that:

```
NAME
SYNOPSIS
       ./coclobas.byte configure [OPTION]...

OPTIONS
       --cluster-name=VAL
           Name of the Kubernetes cluster.

       --cluster-zone=VAL
           Zone of the Kubernetes cluster.

       --help[=FMT] (default=pager)
           Show this help in format FMT (pager, plain or groff).

       --max-nodes=NUMBER
           Maximum number of nodes in th cluster.

       --max-update-errors=VAL (absent=10)
           The number of `kubectl` errors allowed before considering a job
           dead.

       --min-sleep=VAL (absent=180.)
           The maximal time to wait before reentering the update loop.

       --min-sleep=VAL (absent=3.)
           The minimal time to wait before reentering the update loop (events
           like job submission bypass this timer and wake-up the loop any
           way).

       --root=VAL
           The root of the configuration

       --version
           Show version information.

```

Comment (ihodes) [inline src/lib/command_line.ml]: Duh, totally was just testing you… 

:)

**Pull request merged**